src/sdk/person/usePersonQuery.ts

```diff
@@ -0,0 +1,46 @@
+import { useEffect } from 'react'
+import { gql } from '@vtex/graphql-utils'
+import type {
+  PersonQueryQuery,
+  PersonQueryQueryVariables,
+} from '@generated/graphql'
+import { useSession } from '@faststore/sdk'
+
+import { useQuery } from '../graphql/useQuery'
+import type { QueryOptions } from '../graphql/useQuery'
+
+export const query = gql`
+  query PersonQuery {
+    person {
+      id
+      email
+      givenName
+      familyName
+      isAuthenticated
+    }
+  }
+`
+
+const usePersonQuery = (options?: QueryOptions) => {
+  const { data } = useQuery<PersonQueryQuery, PersonQueryQueryVariables>(
+    query,
+    {},
+    { ...options, fetchOptions: { ...options?.fetchOptions, method: 'POST' } }
+  )
+
+  const { setSession, user, ...session } = useSession()
+  const person = data?.person
+
+  useEffect(() => {
+    if (person !== null && person !== undefined && person !== user) {
```

Comment:
```suggestion
    if (Boolean(person) && person !== user) {
```

Why person and user can't be equal? Also, remember the strict inequality (!==) will return true for a different object, even they have the same structure and same values

```js
{} !== {} // true
```

Reply:
Probably you wanna do a shallow equal, right?
Take a look at this shallowEqual from react.
https://github.com/facebook/react/blob/55d75005bc26aa41cddc090273f82aa106729fb8/packages/shared/shallowEqual.js

Reply:
Using `Boolean(person)` it was complaining as it wasn't checking if it was `undefined`, so I changed to check if the value is truth using `!!`, okay?
<img width="801" alt="Screen Shot 2022-03-14 at 11 19 13" src="https://user-images.githubusercontent.com/19983991/158191315-f142f246-1e1e-4a00-a51e-0b345e5eee85.png">

Reply:
> Why person and user can't be equal?
If they are equal then we wouldn't need to update `user`.
> Probably you wanna do a shallow equal, right?
Yeah, you are right! But for some reason it's doing what it should: if I remove the `person !== user` it enters an infinite user-update loop, but if I put this it only updates once - as it should. I don't know exactly why, but since it already interrupts the infinite loop I believe we don't need to implement the shallow equal. What do you think?